PantheonAddonLoader/AddonComponents/Macros.cs

```diff
@@ -28,4 +30,28 @@ public class Macros : IMacros
 
         return match is null ? null : new Macro(match);
     }
-}
\ No newline at end of file
+    public IMacro[]? GetAllMacros()
+    {
+        var macroBar = UIMacroBar.Instance;
+        if (macroBar is null)
+        {
+            return null;
+        }
+
+        var window = macroBar.windowPanel;
+        if (!window.IsVisible)
+        {
+            return null;
+        }
+
+        var buttonRoot = macroBar.ButtonRoot;
+
+        var macroButtons = buttonRoot.GetComponentsInChildren<UIMacroButton>();
+
+        foreach (var macro in macroButtons)
+        {
+            _macros.Add(GetByName(macro.Name));
```

Comment:
```suggestion
            _macros.Add(new Macro(macro));
```
Calling `GetByName` will iterate all macros again, you can create a new macro using the UIMacroButton reference you have here in the loop.

Reply:
Once the other changes are implemented, you can use Linq to return the collection more easily with something like
`return macroButtons.Select(x => new Macro(x));`